### DISCUSSION TITLE
gopsutil use for Fargate managed ECS tasks in AWS

### DISCUSSION THREAD

**Discussion by anandkumar-blr** [shirou/gopsutil#1665] [Q&A]
Hi Guys,
I am working on getting the current CPU and memory usage percentage in the AWS ECS task, specifically looking for task-level usage.
I have used this API and see many differences in values compared to the values we got from container insights at the task level.
When actual Memory (we can see in the Service dashboard ), is 90%, we can see gopsutil API returning around 60%. 
Any help here?
Thanks.

**Comment by shirou** [comment]
We don't know the details of AWS CloudWatch, so we can't say for sure. However, especially regarding memory usage on Linux, the calculation logic is complex. gopsutil [calculates ](https://github.com/shirou/gopsutil/blob/master/mem/mem_linux.go#L307) memory usage as:
```
Total - MemFree - Buffers - (Cached + SReclaimable)
```

If this formula is different, then different values will likely be output.
